utils/gnutls_offset.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+set -e
+
+PROJECT_ROOT_DIR=$(pwd)
+GNUTLS_DIR="${PROJECT_ROOT_DIR}/deps/gnutls"
+OUTPUT_DIR="${PROJECT_ROOT_DIR}/kern"
+
+if [[ ! -f "go.mod" ]]; then
+  echo "Run the script from the project root directory"
+  exit 1
+fi
+
+echo "check file exists: ${GNUTLS_DIR}/.git"
+# skip cloning if the header file of the max supported version is already generated
+if [[ ! -f "${GNUTLS_DIR}/.git" ]]; then
+  echo "check directory exists: ${GNUTLS_DIR}"
+  # skip cloning if the gnutls directory already exists
+  if [[ ! -d "${GNUTLS_DIR}" ]]; then
+    echo "git clone gnutls to ${GNUTLS_DIR}"
+    git clone https://github.com/gnutls/gnutls.git ${GNUTLS_DIR}
+  fi
+fi
+
+
+function run() {
+  git fetch --tags
+  cp -f ${PROJECT_ROOT_DIR}/utils/gnutls_offset.c ${GNUTLS_DIR}/offset.c
+  main_version="3.8"
```

Comment:
Why the later `3.8` version? How to generate the offset addresses of `3.6.*` and `3.7.*`?

Reply:
Perhaps, it is reasonable to only support versions after `3.8.*` to generate offset addresses.

Reply:
need to change `main_version` and loop condition here to support others version, we can do this locally, `3.8.x` is the newest version.